### ISSUE TITLE
query returns still the same result

### ISSUE THREAD

**Issue by zapphyre** [winstonjs/winston-daily-rotate-file#150]
I am using this transport with 'winston-logs-display' module.
This module sends options object into the query method that you normalize and fill it than send it back with callback.
'winston-logs-display' pagintes the results it shows on the page and it's done by increasing the 'start' property of the 'options' object.
Problem is that 'query' method returns still the same result set not taking in account the parameters.

Thanks in advance for solving this!

**Comment by mattberther** [comment]
@zapphyre Thanks for your report. Could you please provide additional information that describes your configuration (node version, transport version, etc) and a small sample that demonstrates the issue?

**Comment by zapphyre** [comment]
$node -v: v8.9.1
"winston-daily-rotate-file": "^3.1.3"

all I did is just pust together winston, daily rotate and winston-logs-display as its supposed to be (i think):
```
const winston = require('winston');
require('winston-daily-rotate-file');

let activityLogTransport = new (winston.transports.DailyRotateFile)({
    filename: 'activity-%DATE%.log',
    datePattern: 'YYYY-MM-DD',
    zippedArchive: false,
    dirname: dirname,
    json: true,
});

require('winston-logs-display')(app, activityLogTransport);
```
I am pleased to see you actually read all the files from the directory with the '%DATE%' pattern but at line 220 in daily-rotate-file.js the callback is called with the same result.
I can see your logic that should go round this on line 199 but it outputs the same result for me..

Hope this helps a bit. 
Thanks!

**Comment by zapphyre** [comment]
I managed to put together a [sample](https://repl.it/@KonicaMinolta/winston-daily-rotate-bugproof).
here you can see, you get two similar arrays even through the second one is shifted by 30 logs (so should not be the same)

here's code (just in case):
```
let winstonDailyRotateFile = require('winston-daily-rotate-file');
let winston = require('winston');

let activityLogTransport = new (winston.transports.DailyRotateFile)({
    filename: 'activity-%DATE%.log',
    datePattern: 'YYYY-MM-DD',
    zippedArchive: false,
    dirname: 'logs',
    json: true,
});

const activityLogger = new (winston.Logger)({
    transports: [
        activityLogTransport
    ],
    exitOnError: false
});

let options = {
      start: 0,
      limit: 30,
      fields: ['reqId']
    }

activityLogger.query(options, (err, logs) => console.log(logs));


options = {
      start: 30,
      limit: 30,
      fields: ['reqId']
    }

activityLogger.query(options, (err, logs) => console.log(logs));
```

**Comment by zapphyre** [comment]
it might be that `processLogFile` keeps calling itself until there is a logfile, but what you want is to return results when the limit provided by options parameter is met.
so how about expanding that condition on line 221 from:
```
if (logFiles.length) {
                processLogFile(logFiles.shift());
}
```
to:
```
if (logFiles.length && results.length < options.limit) {
                processLogFile(logFiles.shift());
}
```

what do you think?

**Comment by mattberther** [comment]
Thank you for your detailed report, @zapphyre. 

This appears to only be an issue when the query options set `order: "desc"`. If you set `order: "asc"` the results are populated appropriately. I will look to revamp the query processing to make it a bit more robust.

**Comment by mattberther** [comment]
@zapphyre I'd like to publish an update version of the transport to address this issue. Please review #152. I've tested it against your repl.it sample and things appear to work.

**Comment by zapphyre** [comment]
It seems to be working, good job!

**Comment by mattberther** [comment]
Resolved with #152 and published to npm as `winston-daily-rotate-file@3.2.0`.
